module-2/backend/week_three.md

```diff
@@ -10,14 +10,65 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+
+  Rails new project_name -T -d postgresql --skip-spring --turbolinks
+  
 2. What do Models generally inherit from in rails?
+    
+    < ApplicationRecord
+  
 3. What do Controllers generally inherit from in a rails project?
+
+  < ApplicationController
+
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+
+  resources :horses, only: [:show]
+
 5. What rake task is useful when looking at routes, and what information does it give you?
+
+  rake routes - shows you what routes you have availave and gives you the prefixes for all your paths
+
 6. What is an example of a route helper? When would you use them?
+
+  horse_new - would take you to the new.erb for making horses
+  used for redirecting for sure; I'm not sure where else? 
+
+
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+
+
+    I'm not sure where they're being entered so I'm not sure what they are returning?   
+    this is a guess - the url is the whole address, whereas path is just the tail end, for instance:  \horses\1
+
 8. What are strong params and why are the necessary?
+  
+    Strong params are passed through a private method that specifies what values are permissible for submission; the private method is only available within the controller class, so it's insulated from other parts of the application. Feel like there's something else here...  
+
 9. What role does `form_for` play in helping us create our forms?
+
+  form_for gives us the ability to capture user-generated input/attributes, submitted to the DB.
+  
 10. How does `form_for` know where to submit the user's input?
+
+ In our view, we enter an instance variable: form_for @horse - which sends us to the controller (? or is @horse working as a route helper ahh! ?)
+ in the controller, we hit the def new method, where @horse = Horse.new, which drops us into the model level, where the fields the user has populated are associated with the attributes we've designated for horse objects. 
+
+
 11. Create a form using a `form_for` helper to create a new `Horse`. 
+
+<h1>Add a new horse for to <%= @owner.name %>'s collection</h1>
+<%= form_for ([@owner, @horse]) do |f| %>
+<%= f.label :name %>
+<%= f.textfield :name %>
+<%= f.label :winnings %>
+<%= f.label :textfield %>
+<%= f.sbumit %>
+<% end %>
+
+I changed this a little because I don't have a great understanding of how @owner is working here and I'd love some illumination; it's the @owner we defined in our controller -- which has a specific id, which in turn allows us to associate the horse with that spefific owner. But I don't understand why/how. 
```

Comment:
The `form_for` uses the arguments to determine the action of the form (this is the route that the form will send its request to). The form above implies that we want our users to be able to create new horses that belong to users. It also implies to me that we want users to be able to create these horses by visiting a different page for each owner (as opposed to selecting an owner from a dropdown or something like that).

We need `@owner` to be passed as part of the argument because the `form_for` is going to use these arguments to create an action like `/owners/4/horses`. The ID in that URI refers to the owner's ID. Without passing `@owner` as an argument, the `form_for` wouldn't know the ID of the owner, and because of that wouldn't be able to generate the `/owners/4/horses` route as an action for the form.

On the other hand, the `form_for` also needs `@horse` as an argument. It uses `@horse` for two reasons: 1) to generate the action on the form (the `/horses` piece of it), and 2) to do some error checking on the contents of the form to make sure that we are only asking users to input information that corresponds to attributes of a Horse as defined in our DB.